src/main.ts

```diff
@@ -1902,18 +1983,18 @@ function generateUnwrap(ctx: Context, fullProtoTypeName: string, fieldNames: Str
         }
       }`);
     } else {
-      chunks.push(code`unwrap(message: Value): string | number | boolean | Object | null | Array<any> | undefined {
-        if (message?.${fieldNames.stringValue} !== undefined) {
+      chunks.push(code`unwrap(message: any): string | number | boolean | Object | null | Array<any> | undefined {
+        if (message?.hasOwnProperty('${fieldNames.stringValue}') && message?.${fieldNames.stringValue} !== undefined) {
```

Comment:
Are the `hasOwnProperty` checks necessary here? Just curious if they are actually fixing something or just being extra.

Reply:
Without it it's looking at the prototype that actually has all other props as `null` for some reason

Reply:
At least in some cases